src/toobj.c

```diff
@@ -949,7 +949,17 @@ void toObjFile(Dsymbol *ds, bool multiobj)
                 return;
 
             Symbol *s = toSymbol(vd);
-            unsigned sz = vd->type->size();
+            d_uns64 sz64 = vd->type->size(vd->loc);
+            if (sz64 == SIZE_INVALID)
+            {
+                vd->error("size overflow");
+                return;
+            }
+            if (sz64 >= 0x1000000)
```

Comment:
I'm not enamored with using a named constant for a single use magic number, because it adds a pointless level of indirection for the cognition of the user.